app/downloader.go

```diff
@@ -26,6 +28,26 @@ func WithThreads(count int) func(*downloader) {
 	}
 }
 
+func WithFilter(arr []string) func(*downloader) {
+	return func(d *downloader) {
+		d.urlSubstrings = arr
+	}
+}
+
+func UrlContains(url *url.URL , needles []string) bool{
```

Comment:
What do you think about supporting globs or regexes?

Reply:
Hi,
As for globs i think that 'contains' is basic '*glob*'. Frankly speaking this is first time I touched golang so tried to keep it simple:) In case of regexes there are  questions:
How to handle "overlapping" patterns?
When to compile regexes (if they are compiled lazy way in go you can hit error after long hours of running lootbox)

I think that (because of daemon-like nature of your tool) nice change preceding regexes would be ability to reload configuration "online", let's say after old fashioned SIGHUP. Also (for me, not every one was so lazy to avoid using regexes;) when you use them there is bigger chance to make mistake, so digging trough tweets 2 hours after you realize that 'smth\s\(' had to be 'smth\(', without ability to reload cofing would not be fun.

Thanks for comments:)
BR.
Artur